src/frontends/lean/info_manager.h

```diff
@@ -62,6 +67,24 @@ class hole_info_data : public info_data_cell {
     pos_info const & get_end_pos() const { return m_end_pos; }
 };
 
+class widget_info : public info_data_cell {
+    environment  m_env;
+    vdom         m_vdom;
```

Comment:
```suggestion
    vdom         m_vdom;
    mutex        m_mutex;
```

I've just been reading through your todo list, and yes, this will be called from multiple threads.  I think the easiest solution for now is to put a `lock_guard<mutex> _(m_mutex)` in both the `report` and `update` functions.

Reply:
Do you know what this error means? 
```
error: binding reference of type ‘std::lock_guard<std::mutex>::mutex_type&’ {aka ‘std::mutex&’} to ‘const std::mutex’ discards qualifiers
  130 |     lock_guard<mutex> _(m_mutex);
      |                         ^~~~~~~
```
Is it because the report function is a const?

Reply:
I'm just going to move const from all the `report`s and see if that solves it

Reply:
I'd prefer to only have a const_cast here.